### PR TITLE
Issue/move profile to header

### PR DESCRIPTION
**Pull request by yuvalron**
see screenshots at #43

### CODE REVIEW

File: src/app/ui/header/header.component.scss [hunk 1]
```diff
@@ -4,6 +4,8 @@
   margin-bottom: 20px;
 
   .container {
-    padding: 10px 0;
+    padding: 5px 0;
   }
 }
+
```
Comment (menhel) [inline src/app/ui/header/header.component.scss]: redundant white spaces.

File: src/app/ui/header/header.component.html [hunk 2]
```diff
@@ -20,19 +23,30 @@
...
+            </div>
+
+            <!-- User NOT logged in -->
+            <div *ngIf="!auth.currentUser">
+              <button class="btn btn-outline-success mr-2" type="button" routerLink="/login">Login</button>
+            </div>
+
           </form>
         </div>
+
```
Comment (menhel) [inline src/app/ui/header/header.component.html]: redundant white space.

File: src/app/ui/header/header.component.html [hunk 3]
```diff
@@ -20,19 +23,30 @@
...
+            <div *ngIf="auth.currentUser">
+              <button class="btn btn-outline-danger mr-2" type="button" (click)="logout()">Logout</button>
+              <img [src]="auth.currentUser.photoURL" width=35px routerLink="/profile">
+            </div>
+
+            <!-- User NOT logged in -->
+            <div *ngIf="!auth.currentUser">
+              <button class="btn btn-outline-success mr-2" type="button" routerLink="/login">Login</button>
+            </div>
+
```
Comment (menhel) [inline src/app/ui/header/header.component.html]: redundant white space.

File: src/app/ui/header/header.component.html [hunk 4]
```diff
@@ -1,14 +1,17 @@
...
-        <a class="navbar-brand" href="#">
-          <img src="/assets/images/testock-logo.png" alt="" height="25">
-        </a>
+
         <button class="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation">
           <span class="navbar-toggler-icon"></span>
         </button>
 
+        <a class="navbar-brand" href="/">
+          <img class="mx-auto d-block" src="/assets/images/testock-logo.png" alt="" height="25">
```
Comment (menhel) [inline src/app/ui/header/header.component.html]: why making this move of the logo?
what is `mx-auto`? what is `d-block`? do you have to use them?

File: src/app/ui/header/header.component.html [hunk 5]
```diff
@@ -20,19 +23,30 @@
             <li class="nav-item">
               <a class="nav-link" routerLink="/courses">Courses</a>
             </li>
-            <!--<li class="nav-item">-->
-              <!--<a class="nav-link" routerLink="/upload">Upload</a>-->
-            <!--</li>-->
             <!--<li class="nav-item">-->
```
Comment (menhel) [inline src/app/ui/header/header.component.html]: remove comments. code should be the cleanest.

File: src/app/ui/header/header.component.html [hunk 6]
```diff
@@ -20,19 +23,30 @@
...
               <!--<a class="nav-link disabled" href="#">ScoreBoard</a>-->
             <!--</li>-->
           </ul>
+
           <form class="form-inline my-2 my-lrg-0">
             <input class="form-control mr-sm-2" type="search" placeholder="Search course..." aria-label="Search">
-            <button class="btn btn-outline-success mr-2" type="submit" routerLink="/courses">Search</button>
-            <button class="btn btn-outline-primary" type="button" routerLink="/upload">Upload scan</button>
+            <button class="btn btn-outline-info mr-2" type="submit" routerLink="/courses">Search</button>
```
Comment (menhel) [inline src/app/ui/header/header.component.html]: If there is more than one button, all of them should be wrapped in a button group.
Wrap all buttons with a div with class `btn-group`.

File: src/app/ui/header/header.component.html [hunk 7]
```diff
@@ -20,19 +23,30 @@
...
             </li>
-            <!--<li class="nav-item">-->
-              <!--<a class="nav-link" routerLink="/upload">Upload</a>-->
-            <!--</li>-->
             <!--<li class="nav-item">-->
               <!--<a class="nav-link disabled" href="#">ScoreBoard</a>-->
             <!--</li>-->
           </ul>
+
           <form class="form-inline my-2 my-lrg-0">
```
Comment (menhel) [inline src/app/ui/header/header.component.html]: This form should only include the search field and search button (other buttons are just links, not part of a form).

File: src/app/users/user-profile/user-profile.component.html [hunk 8]
```diff
@@ -1,14 +1,36 @@
-<!-- User logged in -->
+<div class="col-sm-6 center" id="inner" style="width:35%">
```
Comment (menhel) [inline src/app/users/user-profile/user-profile.component.html]: same issues with col and width as the previous PR

File: src/app/users/user-profile/user-profile.component.html [hunk 9]
```diff
@@ -1,14 +1,36 @@
...
+    </div>
 
-<!-- User NOT logged in -->
-<div *ngIf="!auth.currentUser">
-  <h3>Howdy, {{ auth.currentUserDisplayName }}</h3>
-  <p>Login to get started...</p>
-  <button routerLink="/login">Login</button>
+  </div>
 </div>
+
```
Comment (menhel) [inline src/app/users/user-profile/user-profile.component.html]: Redundant white spaces.

File: src/app/users/user-profile/user-profile.component.scss [hunk 10]
```diff
@@ -0,0 +1,4 @@
+#inner {
```
Comment (menhel) [inline src/app/users/user-profile/user-profile.component.scss]: Same as previous PR.

File: src/app/ui/header/header.component.html (hunk 6 above)
Comment (yuvalron) [reply src/app/ui/header/header.component.html]: without `btn-group` it looks like that: 
image

with `btn-group` it looks like that: 
image

Personally, I prefer the first one.

File: src/app/ui/header/header.component.html (hunk 4 above)
Comment (yuvalron) [reply src/app/ui/header/header.component.html]: return the logo to its original place. 
removed `mx-auto` and `d-block` classes

**Pull request merged**